lib/models/addon-discovery.js

```diff
@@ -20,241 +20,237 @@ var findup     = require('findup');
-  }
-  var discovery = this;
-  var addons = pkg['ember-addon'].paths.map(function(addonPath) {
-    addonPath = path.join(root, addonPath);
-    return discovery.discoverAtPath(addonPath);
-  }, this).filter(Boolean);
-  return addons;
-};
-
-/**
-  Returns a tree based on the internal addons that may be defined within the project.
-  It does this by consulting the projects `supportedInternalAddonPaths()` method, which
-  is primarily used for middleware addons.
-
-  @private
-  @method discoverFromInternalProjectAddons
- */
-AddonDiscovery.prototype.discoverFromInternalProjectAddons = function(project) {
-  var discovery = this;
-  return project.supportedInternalAddonPaths().map(function(path) {
-    return discovery.discoverAtPath(path);
-  }).filter(Boolean);
-};
-
-AddonDiscovery.prototype.discoverFromCli = function (cli) {
-  if (!cli) { return []; }
-
-  var cliPkg = require(path.resolve(cli.root, 'package.json'));
-  return this.discoverInRepoAddons(cli.root, cliPkg);
-};
-
-/**
-  Given a particular path, return undefined if the path is not an addon, or if it is,
-  a node with the info about the addon.
-
-  @private
-  @method discoverAtPath
- */
-AddonDiscovery.prototype.discoverAtPath = function(addonPath) {
-  var pkgPath = path.join(addonPath, 'package.json');
-  debug('attemping to add: %s',  addonPath);
-
-  if (existsSync(pkgPath)) {
-    var addonPkg = require(pkgPath);
-    var keywords = addonPkg.keywords || [];
-    debug(' - module found: %s', addonPkg.name);
-
-    addonPkg['ember-addon'] = addonPkg['ember-addon'] || {};
-
-    if (keywords.indexOf('ember-addon') > -1) {
-      debug(' - is addon, adding...');
-      var addonInfo = {
-        name: addonPkg.name,
-        path: addonPath,
-        pkg: addonPkg,
-      };
-      return addonInfo;
+  },
+
+  /**
+    Returns a tree based on the in-repo addons referenced in the provided `pkg`
+    through paths listed in the `ember-addon` entry, as well as those discovered
+    addons' child addons.
+
+    @private
+    @method discoverInRepoAddons
+   */
+  discoverInRepoAddons: function(root, pkg) {
+    if (!pkg || !pkg['ember-addon'] || !pkg['ember-addon'].paths) {
+      return [];
+    }
+    var discovery = this;
+    var addons = pkg['ember-addon'].paths.map(function(addonPath) {
+      addonPath = path.join(root, addonPath);
+      return discovery.discoverAtPath(addonPath);
+    }, this).filter(Boolean);
+    return addons;
+  },
+
+  /**
+    Returns a tree based on the internal addons that may be defined within the project.
+    It does this by consulting the projects `supportedInternalAddonPaths()` method, which
+    is primarily used for middleware addons.
+
+    @private
+    @method discoverFromInternalProjectAddons
+   */
+  discoverFromInternalProjectAddons: function(project) {
+    var discovery = this;
+    return project.supportedInternalAddonPaths().map(function(path) {
+      return discovery.discoverAtPath(path);
+    }).filter(Boolean);
```

Comment:
this is pre-existing, should likely be fixed in a follow up PR. Enough existing changes here already